src/HotChocolate/Spatial/test/Data.Filters.SqlServer.Tests/FilterVisitorTestBase.cs

```diff
@@ -24,38 +26,33 @@ public FilterVisitorTestBase(PostgreSqlResource<PostgisConfig> resource)
             _resource = resource;
         }
 
-        private Func<IResolverContext, IEnumerable<TResult>> BuildResolver<TResult>(
-            params TResult[] results)
+        private async Task<Func<IResolverContext, IEnumerable<TResult>>>
+            BuildResolverAsync<TResult>(
+                params TResult[] results)
             where TResult : class
         {
-            var dbContext = new DatabaseContext<TResult>(_resource);
+            var databaseName = Guid.NewGuid().ToString("N");
+            var dbContext = new DatabaseContext<TResult>(_resource, databaseName);
 
-            try
-            {
-                Console.WriteLine(results);
-                var sql = dbContext.Database.GenerateCreateScript();
-
-                _resource.RunSqlScriptAsync(sql, "postgis")
-                    .GetAwaiter()
-                    .GetResult();
-                dbContext.AddRange(results);
-                dbContext.SaveChanges();
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.Message);
-            }
+            var sql = dbContext.Database.GenerateCreateScript();
+            await _resource.CreateDatabaseAsync(databaseName);
+            await _resource.RunSqlScriptAsync(
+                "CREATE EXTENSION postgis;\n" + sql,
+                databaseName);
```

Comment:
**nitpick**: we're using the postgis docker image which comes with postgis extension enabled. I don't think we need to run it again as it should be redundant. 
```suggestion
            await _resource.RunSqlScriptAsync(sql, databaseName);
```

Reply:
but ine a new database :)